components/scream/cime_config/buildnml

```diff
@@ -31,68 +39,524 @@ import yaml
 logger = logging.getLogger(__name__)
 
 CIME_VAR_RE = re.compile(r'[$][{](\w+)[}]')
-SWITCH_RE   = re.compile(r'<\s*([^:]+)\s*:')
+METADATA_ATTRIBS = ("type", "valid_values")
+
+###############################################################################
+class MockCase(object):
+###############################################################################
+
+    def __init__(self, kv_dict):
+        self._kv_dict = dict(kv_dict)
+
+    def get_value(self, key):
+        if key in self._kv_dict:
+            return self._kv_dict[key]
+        else:
+            return None
 
 ###############################################################################
 def do_cime_vars(entry, case):
 ###############################################################################
+    """
```

Comment:
Would a more descriptive function name be "dict_entries_to_string"?

Reply:
This function gets values from the CIME Case object, it only looks like a dict lookup because of the MockCase implementation.